### ISSUE TITLE
Suggestion: easier to use = more adoption

### ISSUE THREAD

**Issue by retpoline** [aflnet/aflnet#23]
First, great job on this project so far. Really helping to get more coverage fuzzing lots of servers.

Suggestion I would make it to focus on the aflnet ecosystem and usability. Right now, as I understand it, there are many steps to prep the target for fuzzing. It would be great to narrow these steps into a simple process, closer to what its like for native AFL fuzzing, something like this:

1) Instrument target
2) Patch target
2) Fuzz target

It's not straightforward to develop of course, but its worth the effort. Having a ./install.sh would be helpful as well. "Experimental" automatic patch generation would really speed up the process and less manual effort as well. Try and automate most or all the steps in the process.

Also, more real-world examples instead of just a sample app would be great. Maybe there will be more blogs with demos in time for this in nothing else.

As a reference, the [Mutiny](https://github.com/Cisco-Talos/mutiny-fuzzer) fuzzer has usability in mind: simply feed it a pcap, it generates a .fuzzer file and you pass it to ./mutiny. Super easy and simple for anyone to use.

Just my thoughts (that you all asked for in a different forum :) I hope this is helpful! Awesome project!

**Comment by thuanpv** [comment]
Thanks @retpoline for your suggestions! I really appreciate that!

```
"Experimental" automatic patch generation would really speed up the process and less manual effort as well. Try and automate most or all the steps in the process.
```
This is an interesting point. Could you please elaborate a bit further?

```
Also, more real-world examples instead of just a sample app would be great. Maybe there will be more blogs with demos in time for this in nothing else.
```
Sure. In the tutorials, we have 5 real-world apps and more will be added soon (e.g., OpenSSH, OpenSSL, exim, Kamailio ...). All should come with Docker files to ease the experiments.

```
As a reference, the Mutiny fuzzer has usability in mind: simply feed it a pcap, it generates a .fuzzer file and you pass it to ./mutiny. Super easy and simple for anyone to use.
```
Mutiny looks great!

**Comment by retpoline** [comment]
Yes, the only thing that Mutiny can't do, which is AFLnet's "killer feature" for network fuzzing, is target instrumentation and code coverage.

For patch generation, there probably aren't many public examples of this yet, but the high-level thinking is as follows:

1) Point a component of the AFLnet ecosystem, for example ./aflnet-autopatch at a target file
2) AFLnet has signatures that detect common network functions such as recv(), or accepts an input parameter of "special_app_network_parse()", whatever is desired to be patched for fuzzing and either
2.2) patches the code
or
2.3) generates new code, importing the function and creating a binary to use for easy fuzzing
3) Use AFLnet as usual without having to patch the target code bases yourself

I didn't realize there were already some more examples documented, I'll check them out.

**Comment by thuanpv** [comment]
Hi @retpoline,

FYI, we have just released ProFuzzBench, a benchmark for stateful protocol fuzzing: https://github.com/profuzzbench/profuzzbench. The benchmark includes fuzzing setups for 10 popular protocols and we hope that it could help new users learn how to run experiments and/or support new protocols.

Thuan
